common/src/main/java/com/tc/net/core/TCConnectionImpl.java

```diff
@@ -419,19 +424,25 @@ private boolean buildWriteContextsFromMessages(boolean failfast) {
                 batchMsgCount++;
                 currentBatch.add(batchable);
               } else {
-                this.writeContexts.add(new WriteContext(buildWireProtocolMessage(batchable)));
+                this.writeContexts.add(new WriteContext(buildWireProtocolMessage(batchable), 1));
               }
             } else {
               batchable.complete();
             }
           }
 
-          element = this.writeMessages.poll();
+          try {
+            element = (batchMode) ? this.writeMessages.poll(20, TimeUnit.MICROSECONDS) : this.writeMessages.poll();
+            batchMode = true;
+          } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+            element = null;
+          }
         }
 
         if (MSG_GROUPING_ENABLED && batchMsgCount > 0) {
           final WireProtocolMessage ms = buildWireProtocolMessageGroup(currentBatch);
-          this.writeContexts.add(new WriteContext(ms));
+          this.writeContexts.add(new WriteContext(ms, batchMsgCount));
```

Comment:
In continuation with my previous comment, will the instance of `WriteContext` contain number of messages in batch or total byte size of batch?